draft-ietf-tls-esni.md

```diff
@@ -757,6 +764,15 @@ authenticate the connection with the public name, as follows:
 - If the server requests a client certificate, the client MUST respond with an
   empty Certificate message, denoting no client certificate.
 
+When verifying the certificate, the client uses its application-specific
+certificate profile, e.g. {{Section 4.3.4 of
+?I-D.draft-ietf-httpbis-semantics}}, with a DNS-based reference identity. If
+the public name value is not a valid DNS name for the application,
+the client MUST reject the certificate. In particular, applications that
+incorporate DNS names and IP addresses into the same syntax (e.g. {{?RFC3986,
+Section 7.4}} and {{WHATWG-IPV4}}) MUST reject names that would be interpreted
+as IPv4 addresses.
```

Comment:
```suggestion
?I-D.draft-ietf-httpbis-semantics}}, with the public name as a DNS-based
reference identity. Applications that incorporate DNS names and IP addresses
into the same syntax (e.g. {{?RFC3986, Section 7.4}} and {{WHATWG-IPV4}}) MUST
reject names that would be interpreted as IPv4 addresses.
```

Reply:
(simplifying to further punt on particulars of what certificate validators do with DNS names)

Reply:
Hrm. I guess I was viewing "I think this is actually an IP" as a special case of "I don't accept that DNS name". But maybe that's an unnecessary generalization?

Reply:
Yeah, that's what I'm thinking. Less words overall, and it seems to have the same effect.